### PR TITLE
Apply idea code analysis suggestions Http module

### PR DESCRIPTION
**Pull request by mocenas**
### Summary

Apply code changes from idea to Http module. Mostly fixes unnecessary imports, conversions, throws clause, classes in wrong files etc.

There are also several instances of deprecated APIs, but since that will require more complex changes I want to put that into separate PR.

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [X] Refactoring
- [ ] Backport
- [ ] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [ ] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: http/grpc/src/main/java/io/quarkus/ts/http/grpc/HttpReflectionResource.java [hunk 1]
```diff
@@ -33,13 +33,11 @@ public Uni<GrpcReflectionResponse> reflectionServiceInfo() {
                 .setListServices("").build();
         Uni<ServerReflectionResponse> serverReflectionResponse = invoke(request);
 
-        Uni<GrpcReflectionResponse> grpcReflectionResponse = serverReflectionResponse.map(response -> {
+        return serverReflectionResponse.map(response -> {
```
[the remark this answers is not in the document]
Comment (mocenas) [reply http/grpc/src/main/java/io/quarkus/ts/http/grpc/HttpReflectionResource.java:36]: IMHO this method is short and what is returned is easily readable in method declaration. IMHO that is sufficient. But I also have no strong opinion.

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/MediaTypeResource.java [hunk 2]
```diff
@@ -58,8 +56,7 @@ void addHeaders(final RoutingContext rc) {
     public static class ContentNegotiationContainerResponseFilter implements
             ContainerResponseFilter {
         @Override
-        public void filter(final ContainerRequestContext requestContext, final ContainerResponseContext responseContext)
-                throws IOException {
+        public void filter(final ContainerRequestContext requestContext, final ContainerResponseContext responseContext) {
```
[the remark this answers is not in the document]
Comment (mocenas) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/MediaTypeResource.java]: That makes sense, I will return the exception back.

File: http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/DownloadResourceIT.java [hunk 3]
```diff
@@ -52,7 +50,7 @@ void ensureAsyncFileGetsClosed() throws IOException {
                 .then()
                 .statusCode(204);
 
-        Assert.assertEquals("AsyncFile is not closed, details:\n" + lsofOutput + "\n", 0, lsofOutput.length());
+        Assertions.assertEquals(0, lsofOutput.length(), "AsyncFile is not closed, details:\n" + lsofOutput + "\n");
```
[the remark this answers is not in the document]
Comment (mocenas) [reply http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/DownloadResourceIT.java]: OK

**Review by mocenas — COMMENTED** [review]
Requested changes done.

**Review by gtroitsk — APPROVED** [review]

**Pull request merged**